unit/java_bytecode/generate_concrete_generic_type/generate_java_generic_type.cpp

```diff
@@ -0,0 +1,205 @@
+/*******************************************************************\
+
+ Module: Unit tests for generating new type with generic parameters
+         substitued for concrete types
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+
+#include <util/symbol_table.h>
+
+#include <java_bytecode/generate_java_generic_type.h>
+#include <java_bytecode/java_types.h>
+
+#include <iostream>
+#include <java_bytecode/java_bytecode_language.h>
+#include <language.h>
+
+SCENARIO(
+  "generate_java_generic_type_from_file",
+  "[CORE][java_bytecode][generate_java_generic_type]")
+{
+  std::unique_ptr<languaget> java_lang(new_java_bytecode_language());
+
+  // Configure the path loading
+  cmdlinet command_line;
+  command_line.set(
+    "java-cp-include-files",
+    "./java_bytecode/generate_concrete_generic_type");
+  config.java.classpath.emplace_back(
+    "./java_bytecode/generate_concrete_generic_type");
+
+  // Configure the language
+  null_message_handlert message_handler;
+  java_lang->get_language_options(command_line);
+  java_lang->set_message_handler(message_handler);
+
+  std::istringstream java_code_stream("ignored");
+
+
+  GIVEN("A generic java type with two generic fields and a concrete "
+          "substitution")
+  {
+    java_lang->parse(java_code_stream, "generic_two_fields.class");
+
+    symbol_tablet new_symbol_table;
+    java_lang->typecheck(new_symbol_table, "");
+
+    java_lang->final(new_symbol_table);
+
+    REQUIRE(new_symbol_table.has_symbol(
+      "java::generic_two_fields$bound_element<java::java.lang.Integer>"));
+    THEN("The class should contain two instantiated fields.")
+    {
+      const auto &class_symbol=new_symbol_table.lookup(
+        "java::generic_two_fields$bound_element<java::java.lang.Integer>");
+      const typet &symbol_type=class_symbol.value().get().type;
+
+      REQUIRE(symbol_type.id()==ID_struct);
+      class_typet class_type=to_class_type(symbol_type);
+      REQUIRE(class_type.is_class());
+
+      REQUIRE(class_type.has_component("first"));
+      const auto &first_component=class_type.get_component("first");
+      REQUIRE(is_java_generic_inst_parameter(first_component.type()));
```

Comment:
Could you tighten these conditions up so we check that `first`' and `second`s type is a pointer to an `java.lang.Integer`